src/core/namespace.c

```diff
@@ -3149,6 +3151,104 @@ bool ns_type_supported(NamespaceType type) {
         return access(ns_proc, F_OK) == 0;
 }
 
+int refresh_extensions_in_namespace(PidRef *target, const char *hierarchy_env, NamespaceParameters *p) {
+        const char *overlay_prefix = "/run/systemd/mount-rootfs";
+        _cleanup_(mount_list_done) MountList ml = {};
+        _cleanup_free_ char *extension_dir = NULL;
+        _cleanup_strv_free_ char **hierarchies = NULL;
+        int r;
+
+        assert(target);
+        assert(hierarchy_env);
+        assert(p);
+
+        log_debug("Refreshing extensions in-namespace for hierarchy '%s'", hierarchy_env);
```

Comment:
If we are only expecting reloads for CONFEXT type hierarchy, could it make sense to keep the env name hardcoded and do the parse operation on just the CONFEXT hierarchy type?

Reply:
The hardcoding here is done in `service.c` to allow for easier future expansions to other types if it's helpful.